openshift/generate-manifests.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Know where the repo root is so we can reference things relative to it
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+
+# Source bingo so we can use kustomize and yq
+. "${REPO_ROOT}/.bingo/variables.env"
+
+# We're going to do file manipulation, so let's work in a temp dir
+TMP_ROOT="$(mktemp -p . -d 2>/dev/null || mktemp -d ./tmpdir.XXXXXXX)"
+# Make sure to delete the temp dir when we exit
+trap 'rm -rf $TMP_ROOT' EXIT
+
+# Copy all kustomize files into a temp dir
+TMP_CONFIG="${TMP_ROOT}/config"
+cp -a "${REPO_ROOT}/config" "$TMP_CONFIG"
+
+# Override namespace to openshift-catalogd
+$YQ -i '.namespace = "openshift-catalogd"' "${TMP_CONFIG}/default/kustomization.yaml"
+
+# Create a temp dir for manifests
+TMP_MANIFEST_DIR="${TMP_ROOT}/manifests"
+mkdir -p "$TMP_MANIFEST_DIR"
+
+# Run kustomize, which emits a single yaml file
+TMP_KUSTOMIZE_OUTPUT="${TMP_MANIFEST_DIR}/temp.yaml"
+$KUSTOMIZE build "${TMP_CONFIG}/default" -o "$TMP_KUSTOMIZE_OUTPUT"
+
+# Use yq to split the single yaml file into 1 per document.
+# Naming convention: $index-$kind-$namespace-$name. If $namespace is empty, just use the empty string.
+(
+  cd "$TMP_MANIFEST_DIR"
+
+  # shellcheck disable=SC2016
+  yq -s '$index +"-"+ (.kind|downcase) +"-"+ (.metadata.namespace // "") +"-"+ .metadata.name' temp.yaml
```

Comment:
Just found this while making a version for rukpak:
```suggestion
  $YQ -s '$index +"-"+ (.kind|downcase) +"-"+ (.metadata.namespace // "") +"-"+ .metadata.name' temp.yaml
```